panel/src/panel/content.js

```diff
@@ -149,6 +159,30 @@ export default (panel) => {
 		 */
 		get values() {
 			return panel.view.props.content;
+		},
+
+		rescue() {
+			if (this.isSaved === false) {
+				const data = new FormData();
+
+				for (const field in panel.view.props.content) {
+					data.append(field, panel.view.props.content[field]);
+				}
+
+				navigator.sendBeacon(panel.url(this.api + "/changes/save"), data);
```

Comment:
Did you know how much data we can send via sendBeacon? Does it have any limits? I've never worked with it before.

Reply:
From various discussions on the web, it's depending on the browser but seems to be limited in Chrome and Safari to 64 KB. Same for `fetch` with `keepalive`, which though isn't supported by Firefox yet.

Alternatively, the preventing `beforeunload` event (which opens this "you really want to close, there might be unsaved suer data" popup) would be nicer in the way that it jus buys enough time to send the last regular throttled request. But the message is not configurable, so it can be confusing to the user, e.g. thinking that they need to really save their unsaved changes and not just wait a second for storing unsaved changes.

We could still use this as the safer alternative in terms of data processing, but maybe reduce the throttle delay to decrease the likelihood of that opoup needing to show up ever.

Reply:
Is it complicated to implement the popup? If it's easy to do, I would probably really go this way as a first solution and then see how often we run into it and how confusing it really is.

Reply:
It's super easy:

```js
window.addEventListener('beforeunload', function (e) {
    e.preventDefault();
    e.returnValue = '';
});
```

Or for our case then inside wrapped with `if (this.isSaved === false)`.

Reply:
Then I would really go this way - at least for the alphas and betas.